lib/cinematic.ts

```diff
@@ -434,6 +434,16 @@ class Cinematic {
             return;
         }
 
+        let currentVideo = this.playlist.getCurrentVideo();
+        let newSource = currentVideo.getSourcesForQuality(newQuality);
+        if (!newSource) {
+            newQuality = currentVideo.getBestAvailableQuality();
+            newSource = currentVideo.getSourcesForQuality(newQuality);
+        }
+        if (!newSource) {
+            return;
+        }
```

Comment:
```suggestion
        if (!newSource) {
            newQuality = currentVideo.getBestAvailableQuality();
        }
        if (!newSource = currentVideo.getSourcesForQuality(newQuality);) {
            return;
        }
```

this should work too, right? maybe this is more understandable?

Reply:
I would be careful with actual commit suggestions in the TypeScript repos as applying them will not update the compiled js and map files.

Reply:
It should to the same but in my optinion assigments in if clauses can potentially be pretty dangerous as it looks like the author did the comparison wrong and actually meant `==` or `===` when coming from other languages. "Fixing" it then breaks the code.